autopilot/hostscore.go

```diff
@@ -237,13 +245,22 @@ func uptimeScore(h api.Host) float64 {
 	return math.Pow(ratio, 200*math.Min(1-ratio, 0.30))
 }
 
-func versionScore(settings rhpv2.HostSettings) float64 {
+func versionScore(settings rhpv2.HostSettings, minVersion string) float64 {
+	if minVersion == "" {
+		minVersion = minProtocolVersion
+	}
 	versions := []struct {
 		version string
 		penalty float64
 	}{
-		{"1.6.0", 0.99},
-		{"1.5.9", 0.00},
+		// latest protocol version
+		{"1.6.0", 0.10},
+
+		// user-defined minimum
+		{minVersion, 0.00},
+
+		// absolute minimum
+		{minProtocolVersion, 0.00},
```

Comment:
if the `minProtocolVersion` constant is an absolute minimum we should not allow setting a lower user-defined minimum and add it to whatever validation we have on those settings

Reply:
I thought about that but we can't guarantee consistency since we might raise the absolute minimum on our end. So if the user sets 1.6.0 and we raise it from 1.5.9 to 1.6.1 it might be confusing since the user has a setting that they couldn't possibly set anymore.